packages/playwright/src/worker/testInfo.ts

```diff
@@ -481,20 +481,29 @@ export class TestInfoImpl implements TestInfo {
     return this._resolveSnapshotPath(undefined, legacyTemplate, pathSegments);
   }
 
-  skip(...args: [arg?: any, description?: string]) {
-    this._modifier('skip', args);
+  skip = wrapFunctionWithLocation((location: Location, ...args: [arg?: any, description?: string]) => this._skipWithLocation(location, ...args));
+
+  fixme = wrapFunctionWithLocation((location: Location, ...args: [arg?: any, description?: string]) => this._fixmeWithLocation(location, args));
+
+  fail = wrapFunctionWithLocation((location: Location, ...args: [arg?: any, description?: string]) => this._failWithLocation(location, args));
+
+  slow = wrapFunctionWithLocation((location: Location, ...args: [ arg?: any, description?: string]) => this._slowWithLocation(location, args));
+
+  // These are called with dynamic dispatch from the `TestTypeImpl` and `WorkerMain` classes
+  _skipWithLocation(location: Location, ...args: [arg?: any, description?: string]) {
```

Comment:
Would like to know a nicer way to handle this. Methods in `testType` and `workerMain` will call `testInfo[`_${type}WithLocation`]` (previously `testInfo[type]`) dynamically, but we don't want them to generate their own location.